text/0033-fake-account-packet/0033-fake-account-packet.md

```diff
@@ -0,0 +1,206 @@
+pub struct ExampleAccount {
+    pub id: u64,
+    pub other_stuff: Vec<u8>,
+}
+
+impl Fake for ExampleAccount {
+    fn fake(seed: &Seed) -> ExampleAccount {
+        let mut rng = IsaacRng::from_seed(&Self::to_rng_seed(seed));
+        let size: u8 = rng.gen();
+        ExampleAccount {
+            id: rng.gen(),
+            other_stuff: rng.gen_iter().take(size as usize).collect(),
+        }
+    }
+
+    fn fake_version(_seed: &Seed) -> u64 {
+        2
+    }
+}
+```
+
+and can be used in [`create_fake_account_packet()`]
+(https://gitlab.com/Fraser999/Fake-Account-Packet/blob/master/src/main.rs#L55-72):
+
+```rust
+fn create_fake_account_packet(id: XorName, session_ids: &[SessionId]) -> StructuredData {
+    let seed = get_seed(&id, &session_ids);
+    let (public_key, private_key) = sign::keypair_from_seed(&seed);
+    let type_tag = 0;
+    let fake_account = ExampleAccount::fake(&seed);
+    let data = serialisation::serialise(&fake_account).expect("Can't fail.");
+    let version = ExampleAccount::fake_version(&seed);
+    StructuredData::new(type_tag,
+                        id,
+                        version,
+                        data,
+                        vec![public_key],
+                        vec![],
+                        Some(&private_key))
+        .expect("Can't fail.")
+}
+```
+
+# Drawbacks
+
+This presents a very basic implementation.  It requires the Vaults to generate throwaway
+cryptographic keys in order to create a plausible fake - a relatively expensive operation.  For an
+attacker just looking to do a DOS attack, this might provide a vector.  Profiling may show that this
+proposed defence may need to become more complex, e.g. monitoring and reacting to unusually high
+rates of `Get` requests for non-existent data, or caching spare throwaway crypto keys during quiet
+periods for later use in generating fakes.
+
+# Alternatives
+
+1. Don't respond to `Get` requests for non-existent data.  While this would be simple to implement
+in Vaults and would be the optimal way to avoid allowing an attacker to make Vaults "do work", it
+causes problems for legitimate Clients which request a non-existent account.  Clients currently are
+guaranteed to receive a response to every request.  If this changes, Clients would need to take
+extra care to not ask for non-existent data, or else handle a non-response (possibly through a
+timeout).
+
+2. Respond with `GetFailure` for non-existent data, but monitor the rate at which such requests are
+passing by.  This would need to be handled by the Vault through which the Client is connected.  Once
+an unacceptably high rate of `GetFailure`s is reached, the Client is dropped.  This seems fairly
+simple to implement, but an attacker would be able to work around this by using many Clients.
+
+# Unresolved questions
+
+1. Should the seed be generated using all members' names from the close group, or just a subset of
```

Comment:
Yup - agreed.  This would only reduce the chance of churn affecting the seed, but not eliminate it.  And in theory we could as equally well choose the furthest members of the close group and achieve the same level of resilience to churn.

I'm happy to leave it as the whole group being used though, if there are no objections to that.